content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,367 @@
+This is the transitive closure property that this type of systems typically use to fight against Sybil attacks.
+
+The overlapping nodes could be all Byzantine nodes.
+In that case, having quorum intersection does not help the network to agree at all.
+For that reason, the most important assumption about quorum intersection model is that the network enjoys quorum intersection `afer` all misbehaving nodes are removed from the network.
+
+A living example of such a network is internet itself. It consists of independent nodes that know about few local to them, but the sets overlap with each other and every node is reachable through any other by routes and gateways.
+In the same manner, FBA network resembles inter-domain network routing with no central authority dictating or arbitraging agreements.
+We can think of quorum slices as analogius to internet networks in terms of reachability.
+Servers in the internet networks can be reached via requests from neighboring networks, using gateways and hubs, making each server accessible through any client.
+In quorums, nodes are reachable through their neighboring nodes, that might be in another quorum, but because of the nodes connections between each other, each node can be reached theoretically from any other.
+The Internet's nearly complete transitive reachability suggests it can likewise ensure a worldwide consensus with FBA.
+
+Nodes can be categorized as either well-behaved or ill-behaved.
+A well-behaved node chooses sensible quorum slices and obeys the protocol, including eventually responding to all requests.
+An ill-behaved nodes does not.
+Ill-behaved nodes suffer Byzantine failure, meaning they can behave arbitrarily.
+Since the blockchains network most important goal is to reach consensus deterministically — meaning reach to the same consensus value every time in the same circumstances, nodes which are generally acting arbitrarily (nondeterministically) cannot be considered functioning parts of the system.
+Expected behavior of well-behaved nodes is always `deterministic` in the sense that they choose the same right value among others every time.
+
+The goal of Byzantine agreement is to ensure that well-behaved nodes externalize the same values despite the presence of the ill-behaved nodes.
+The first part of this goal is to prevent nodes from diverging and externalizing different values for the same slot.
+Second part is to ensure the nodes can externalize values practically always, as opposed to getting blocked in some dead-state.
+Corresponding to these properties are notions of safety and liveness.
+
+TODO: add a picture of voting steps
+
+#### Stellar
+
+One of the most famous consensus protocol proposed, that uses Avalanche and FBA to build decentralized networks is [Stellar Consensus Protocol (SCP)]((https://www.stellar.org/papers/stellar-consensus-protocol?locale=en)), that was described in 2015.
```

Comment:
`consensus protocols`

I don't get the part about using Avalanche? Stellar doesn't use Avalanche at all.

Reply:
Yes, I missed this in the text; removed it from Stellar text;
https://github.com/optakt/optakt.github.io/pull/21/commits/ca80e6c74bee64b8ce712ca515bb205ee645d971#diff-4bf8323c6426a93da1ce22be1b8528779ef6b61fdacef3e0d065f1e1dd47bea5R102
Thanks!